sklearn/neighbors/tests/test_nca.py

```diff
@@ -129,7 +129,6 @@ def test_params_validation():
     # TypeError
     assert_raises(TypeError, NCA(max_iter='21').fit, X, y)
     assert_raises(TypeError, NCA(verbose='true').fit, X, y)
-    assert_raises(TypeError, NCA(tol=1).fit, X, y)
     assert_raises(TypeError, NCA(n_components='invalid').fit, X, y)
```

Comment:
integers are part of `numbers.Real`. But afaik there's not type for 'any float type but not int type'.
However, I don't see why setting tol=1 would be bad.